tests/test_reification.py

```diff
@@ -33,118 +31,142 @@ def parse_program(stm, parameters=[], name="base"):
     )
     return flatten(ret)
 
-
 def clingo_parse_program(stm):
     ret = []
     ast.parse_string(stm, ret.append)
-    ret = [str(rule) for rule in ret]
+    ret = [rule for rule in ret]
     return ret
 
-
-class TestCase(unittest.TestCase):
-
+class TestCase(ASTTestCase):
     def setUp(self):
         self.print = False
 
     def assert_equal_program(self, program, expected):
         expected_program = clingo_parse_program(expected)
-        self.assertListEqual(sorted(program), sorted(expected_program))
-
+        expected_program = function_transformer._rule_to_symbolic_term_adapter(expected_program)
+        
+        #print("\nThe expected program\n", expected_program)
+        #print("\nThe parsed program\n", program)
+        
+        
+        sorted_program = sorted(program)
+        expected_program.sort()
```

Comment:
But, now I'm trying to figure out some issues for the asserting case of the lists being equal. While they are equal, they are being ordered in the wrong order and that is causing tests to fail. I'll look into it tomorrow